File: Assets/Scripts/Cards/CardInstance.cs
namespace DokkaebiHand.Cards
{
    /// <summary>
    /// 게임 내에서 사용되는 카드 인스턴스.
    /// ScriptableObject 참조 또는 데이터베이스 정의를 래핑.
    /// </summary>
    public class CardInstance
    {
        public int Id { get; private set; }
        public string Name { get; private set; }
        public string NameKR { get; private set; }
        public CardMonth Month { get; private set; }
        public CardType Type { get; private set; }
        public RibbonType Ribbon { get; private set; }
        public int BasePoints { get; private set; }
        public bool IsRainGwang { get; private set; }
        public bool IsDoublePi { get; private set; }

        public CardInstance(int id, HwaTuCardDatabase.CardDefinition def)
        {
            Id = id;
            Name = def.Name;
            NameKR = def.NameKR;
            Month = def.Month;
            Type = def.Type;
            Ribbon = def.Ribbon;
            BasePoints = def.BasePoints;
            IsRainGwang = def.IsRainGwang;
            IsDoublePi = def.IsDoublePi;
        }

        public int GetPiValue()
        {
            if (Type == CardType.Pi)
                return IsDoublePi ? 2 : 1;
            return 0;
        }

        public override string ToString()
        {
            return $"[{Id}] {NameKR} ({Month}월 {Type})";
        }
    }
}

File: dokkaebi-love2d/src/cards/card_instance.lua
--- 카드 런타임 인스턴스
local Enums = require("src.cards.card_enums")

local CardInstance = {}
CardInstance.__index = CardInstance

function CardInstance.new(id, def)
    return setmetatable({
        id = id,
        name = def.name,
        name_kr = def.name_kr,
        month = def.month,
        card_type = def.card_type,
        ribbon = def.ribbon or Enums.RibbonType.None,
        base_points = def.base_points or 1,
        is_rain_gwang = def.is_rain_gwang or false,
        is_double_pi = def.is_double_pi or false,
        pi_index = def.pi_index or 1,
    }, CardInstance)
end

function CardInstance:get_pi_value()
    if self.card_type == Enums.CardType.Pi then
        return self.is_double_pi and 2 or 1
    end
    return 0
end

function CardInstance:__tostring()
    return string.format("%d월 %s (%s)", self.month, self.name_kr, self.card_type)
end

return CardInstance
